### DISCUSSION TITLE
How many threads does Pluto use vs `Threads.nthreads()`?

### DISCUSSION THREAD

**Discussion by adomasbaliuka** [fonsp/Pluto.jl#2087] [Questions about using Pluto] [answered]
If I start Julia with `--threads=1`  (or `--threads=6`) , running `Threads.nthreads()` in the console gives 1 (or 6), as expected. If I then use the Julia instance to start a Pluto notebook, `Threads.nthreads()` **in the notebook always gives 4**. Why is this? Does it have to do with the browser? Is it representative of how many threads the notebook can actually use and relevant for performance?

```
Julia Version 1.7.1
Commit ac5cc99908 (2021-12-22 19:35 UTC)
Platform Info:
OS: Linux (x86_64-pc-linux-gnu)
CPU: Intel(R) Core(TM) i7-8565U CPU @ 1.80GHz
WORD_SIZE: 64
LIBM: libopenlibm
LLVM: libLLVM-12.0.1 (ORCJIT, skylake)
Environment:
JULIA_REVISE_WORKER_ONLY = 1
```

**Comment by rikhuijzer** [comment]
It's related to this code:

https://github.com/fonsp/Pluto.jl/blob/f03f73ca2e345dfb3138521bcf48ac8c06250390/src/Configuration.jl#L153-L170

Why this is the case only Fons knows.

Can you try overriding it by setting the environment variable `JULIA_NUM_THREADS` instead of the startup flag? Does that change anything?

**Comment by fonsp** [comment, marked as answer]
The flags that you give to `julia` only apply to Pluto's *server*, not to the notebook processes that it spawns. So if you run `Threads.nthreads()` in a notebook, it is giving you the number of threads of the notebook process, not the server process.

You can set parameters for notebook processes with keyword arguments to `Pluto.run`, e.g.

```julia
Pluto.run(threads="16", compile="0")
```
